src/transaction.js

```diff
@@ -69,6 +97,12 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
     })
   }
 
+  if (hasWitnesses) {
+    for (i = 0; i < vinLen; ++i) {
+      tx.inputs[i].witness = readScript()
```

Comment:
Re above & witness format:

    0x03 # number of witnesses/inputs
      00 # say the first has no data elements in the witness
      0x02 0x41 [pubkey] 0x.... [sig] # this one has two elements. each element is prefixed with it's length, but using a varint(not a script push)
      0x01 0x00 # this input has a witness. it contains one element, a byte of \x00